tsdb/engine/tsm1/reader.go

```diff
@@ -895,7 +895,7 @@ func (d *indirectIndex) OverlapsTimeRange(min, max int64) bool {
 
 // OverlapsKeyRange returns true if the min and max keys of the file overlap the arguments min and max.
 func (d *indirectIndex) OverlapsKeyRange(min, max string) bool {
-	return d.minKey <= max && d.maxKey >= max
+	return d.minKey <= max && d.maxKey >= min
```

Comment:
This looks like it could have been the source of other bugs unrelated to the dropping/missing data issue. Do we have any other open issues that could be hitting this, and how do you think they would manifest themselves?